setup.py

```diff
@@ -341,18 +345,17 @@ def get_extensions():
     else:
         image_src += glob.glob(os.path.join(image_path, "cuda", "*.cpp"))
 
-    if use_png or use_jpeg:
-        ext_modules.append(
-            extension(
-                "torchvision.image",
-                image_src,
-                include_dirs=image_include + include_dirs + [image_path],
-                library_dirs=image_library + library_dirs,
-                define_macros=image_macros,
-                libraries=image_link_flags,
-                extra_compile_args=extra_compile_args,
-            )
+    ext_modules.append(
```

Comment:
We are now always building the `image` library even if png and jpeg haven't been found, since we can always build libgif anyways. Maybe we'll add a `WITH_LIBGIF` flag or something similar, but this can come later.

Reply:
Would be nice to add a `TORCHVISION_USE_GIF` env var similar to all other image backends for consistency.

Reply:
Curious why you'd need it @adamjstewart ?
Those are useful for the other libraries as we dynamically link against them and they introduce dependencies, but since we vendor libgif we don't have the same issues with the gif decoder.

Reply:
I'm coming from the perspective of a maintainer of a from-source package manager: [Spack](https://spack.io). We need to be able to build packages like torchvision from source on all kinds of weird systems (Linux aarch64, ppc64le, etc.) and with compilers (OneAPI, AOCC, Fujitsu, etc.) for which they are not widely tested. This requires patching these packages to fix compilation errors. For example, our giflib recipe already has multiple patches required for certain systems. Vendoring results in us having to patch multiple locations instead of one. I'm not opposed to it being optionally vendored, but would love support for externally-installed copies too.

Reply:
it would be useful to have an option to use the shared library. We maintain things at conda-forge for shared linking which would make this easier to integrated.

One bug i'm running into (with the current startegy) is that clang for example likes to complain that `-std-c++17` is added to compilation of a c file....

Reply:
Adam, you can likely take my small patches to unvendor 
https://github.com/conda-forge/torchvision-feedstock/pull/94

Reply:
they would definitely need more work to be submitted here